### PR TITLE
[SDM] Extracts webpage loading time for training set

### PR DESCRIPTION
**Pull request by Chikdolman**
The code extracts loading time and stores in result.csv.
The extracted loading time is used for **service discovery training set.**
As shown in the weightFinder/renderer/server.py, this code modifies the HW parameter of the renderer.

### CODE REVIEW

File: service_discovery_manager/Tools/weightFinder/browser/loadingtime_extract.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+//To get loading time using node.js
+//After opening chrome browser with option --remote-debugging-port=9222
+//usage: node ./filename.js http://www.url.com
+//
+const CDP = require('chrome-remote-interface');
+const fs = require('fs');
+const util = require('util');
+const path = require('path');
+
+const g_duration = 20*1000
```
Comment (younghajung) [inline service_discovery_manager/Tools/weightFinder/browser/loadingtime_extract.js]: Need a space among operand and operator.

File: service_discovery_manager/Tools/weightFinder/renderer/server.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+def setNetworkBandwidth(packetDropRate): #since only egress can be controlled and ingress cannot be controled, we control the bandwidth with packet drop rate.
+    print "Set packetDropRate: " + str(packetDropRate)
+    cmd = 'tc qdisc replace dev enp4s0 parent 1:0 handle 10: netem loss ' + str(packetDropRate)+'%'
+    print cmd
+    os.system(cmd)
+    return
+'''
+
+def setCPUUtilization(utilization):
+    utilization = (100 - utilization)*g_numCores # limit cpu usage (Current CPU utilization 30% = my proc can use maximum 70% of the CPU)
```
Comment (younghajung) [inline service_discovery_manager/Tools/weightFinder/renderer/server.py]: Need a space among operand and operator.

File: service_discovery_manager/Tools/weightFinder/browser/runAutomationTool.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import sys, csv, socket, json
+from subprocess import check_output
+
+global g_ip
+global g_port
+
+global g_socket
+
+g_loadingTimeExtractor = './loadingtime_extract.js'
+g_availableFreq = [3200, 2500, 1800, 1100, 800]
```
Comment (younghajung) [inline service_discovery_manager/Tools/weightFinder/browser/runAutomationTool.py:14]: g_availableFreq is not matched renderer's g_availableFreq. Is it OK?

File: service_discovery_manager/Tools/weightFinder/browser/loadingtime_extract.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+CDP(async (client) => {
+	try {
+		const {Page} = client;
+		const loadEventFired = Page.loadEventFired();
+
+		await Page.enable();
+		for(i = 0; i < numRun; i++){
+			start_time = new Date();
+			await Page.navigate({url: args[0]});
+			//await Page.navigate({url: 'https://www.naver.com'});
```
Comment (younghajung) [inline service_discovery_manager/Tools/weightFinder/browser/loadingtime_extract.js]: Please remove all commented codes.

File: service_discovery_manager/Tools/weightFinder/browser/runAutomationTool.py (hunk 3 above)
Comment (Chikdolman) [reply service_discovery_manager/Tools/weightFinder/browser/runAutomationTool.py:14]: Yes it is intentional. This reduces number of tests which had been too many tests before.

File: service_discovery_manager/Tools/weightFinder/browser/loadingtime_extract.js (hunk 1 above)
Comment (Chikdolman) [reply service_discovery_manager/Tools/weightFinder/browser/loadingtime_extract.js]: Done

File: service_discovery_manager/Tools/weightFinder/renderer/server.py (hunk 2 above)
Comment (Chikdolman) [reply service_discovery_manager/Tools/weightFinder/renderer/server.py]: Done

**Review by younghajung — APPROVED** [review]
LGTM

**Pull request merged**